custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -7,18 +8,114 @@ primary_entity:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF      
     - id: 2
       name: speed
       type: string
+      hidden: true
       mapping:
         - dps_val: large
           value: 100
+          icon: "mdi:sprinkler-variant"
         - dps_val: small
           value: 50
+          icon: "mdi:sprinkler"
+    - id: 4          
+      name: "Left time"
+      readolny: true
+      optional: true
+      type: integer
+      range:
+        min: 0
+        max: 360
 secondary_entities:
   - entity: light
     name: Nightlight
+    icon: "mdi:light-recessed"
     dps:
       - id: 5
         name: switch
         type: boolean
+      - id: 6 
+        name: color_mode  #effect
+        type: string
+        hidden: true
+        mapping:
+          - dps_val: colour
+            value: rgbw 
+      - id: 8   # not working
+        name: rgbhsv  
+        type: hex     
+        format:
+          - name: r
+            bytes: 1
+          - name: g
+            bytes: 1
+          - name: b
+            bytes: 1
+          - name: h
+            bytes: 2
+            range:
+              min: 0
+              max: 360
+          - name: s
+            bytes: 1
+            range:
+              min: 0
+              max: 100
+          - name: v
+            bytes: 1
+            range:
+              min: 0
+              max: 100
+  - entity: select
+    name: "Color mode"
+    category: config
+    icon: "mdi:palette"
+    dps:
+    - id: 6
+      name: option
+      type: string
+      mapping:
+        - dps_val: colour
+          value: Color
+        - dps_val: colourful1
+          value: Colorful          
+  - entity: select
+    name: "Spray mode"
+    category: config
+    icon: "mdi:sprinkler-variant"
+    dps:
+    - id: 2
```

Comment:
Indentation looks wrong again.  But this is already covered by the fan speed, so redundant.

Reply:
Since this is not fan I find this approach better. This way simple button can be used and it looks much better.